### DISCUSSION TITLE
v5.49.0

### DISCUSSION THREAD

**Discussion by sokra** [webpack/webpack#13945] [General]
# Features

* add `experiments.buildHttp` to build `http(s)://` imports instead of keeping them external
  * keeps a `webpack.lock` file with integrity and `webpack.lock.data` with cached content that should be committed
  * Automatically upgrades lockfile during development when remote resources change
    (might be disabled with `experiments.buildHttp.upgrade: false`)
  * Lockfile is frozen during production builds and usually no network requests are made
    (exception: `Cache-Control: no-cache`).
  * The `webpack.lock.data` persisting can be disabled with `experiments.buildHttp.cacheLocation: false`.
    That will will introduce a availability risk.
    (webpack cache will be used to cache network responses)

# Bugfixes

* fix HMR infinite loop (again)
* fix rare non-determinism with `splitChunks.maxSize` introduces in the last release
* optional modules no longer cause the module to fail when `bail` is set
* fix typo in records format: chunkHashs -> chunkHashes

# Performance

* limit the number of parallel generated chunks for memory reasons

<hr /><em>This discussion was created from the release <a href='https://github.com/webpack/webpack/releases/tag/v5.49.0'>v5.49.0</a>.</em>

**Comment by shockw4ver** [comment]
Save my day!!! 😂... 
Why Can't I found any discussions about HMR infinite loop 😭...

**Comment by wangyan19891215** [comment]
The image introduced in CSS will be generated with a hash value of.jpg after packaging. This image cannot be displayed. The hash image referenced in CSS after packaging will not be generated in Webpack-v5.47.1, so this problem is not solved  
![image](https://user-images.githubusercontent.com/55070404/130710535-0f5e9ecf-aa66-43a3-9745-fc74eefe07a7.png)

**Comment by wangyan19891215** [comment]
![image](https://user-images.githubusercontent.com/55070404/130710629-77e429a7-8a89-456d-a79e-aba2e4c7707d.png)
